### PR TITLE
[WIP] test/performance: initial commit of gimbal performance test script

### PR DESCRIPTION
**Pull request by alexbrand**
Initial work on script for automating performance testing. Decided to use go for this, as opposed to scripting it out in bash. The main reason is that we have client-go, which enables us to perform kubernetes operations more precisely vs. kubectl + grep.

Some things I want to improve over time:
- [x] Improve storage of logs. Maybe timestamp them, or create directories for each test.
- [x] Split out funcs into separate files where it makes sense.
- [ ] Use a configuration file instead of flags for passing config (maybe?)
- [x] Use IngressRoute CRD instead of Ingress
- [ ] Add openstack discovery time test

Added vendored dependencies as a separate commit.

### CODE REVIEW

**Comment by rosskukulinski** [comment]
Refs #72

File: test/performance/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+### Run tests
+
+```sh
+make
+./gimbalbench \
+    --gimbal-kubecfg-file ${GIMBAL_KUBECONFIG} \
+    --backend-kubecfg-file ${BACKEND_KUBECONFIG} \
+    --loadgen-kubecfg-file ${LOADGEN_KUBECONFIG} \
+    --gimbal-url ${GIMBAL_URL} \
+    --concurrent-connections-test 250,500,1000,2500 \
```
Comment (stevesloka) [inline test/performance/README.md]: What do these values mean? May document/link to how I would customize?

File: test/performance/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    --backend-services-test 10,25,50,100 \
+    --backend-endpoints-test 10,25,50,100
+```
+
+### Test results
+
+The wrk2 results are stored in the `logs` directory.
```
Comment (stevesloka) [inline test/performance/README.md]: Does each run get its own directory? i.e. can I run this over and over and have the logs for each run stored?

**Comment by stevesloka** [comment]
I think overall this looks great! I'd like to shadow you using it next time tests are run to get a feel for the inputs/outputs, etc.

**Comment by rosskukulinski** [comment]
This is great!

@stevesloka / @alexbrand I think it would be good to do a zoom recording of running through the tests.  I believe Actapio is interested in using this test suite in their environment, so any additional documentation (maybe a diagram too?) would be helpful.

Alex: You made a list of things you'd like to improve over time -- I suggest writing separate issues for each one and label with `testing` issue label.  Once you've done that, update this PR discussion with the issue links.

**Comment by alexbrand** [comment]
Added some initial work to generate a report from the test results. The report can then be uploaded into a spreadsheet in Google sheets. The readme has been updated with new instructions.

**Comment by alexbrand** [comment]
Added a diagram and updated README.

**Comment by rosskukulinski** [comment]
@alexbrand I think we should move the perf testing work to it's own repo.  Closing this PR.

**Pull request closed without merging**